### PR TITLE
Added priority adjustment limits (DB-19576)

### PR DESCRIPTION
**Pull request by susmita-saha**
Added priority adjustment limits
* ADJUSTFACTOR: priority_adjustment_factor (double)
* ADJUSTTIME: priority_adjustment_time (int)

````
SET ADJUSTTIME 15;
SET ADJUSTFACTOR "-0.25";
-----------------------------
SET ADJUSTTIME RESET;
SET ADJUSTFACTOR RESET;
````

### CODE REVIEW

File: src/main/java/com/ocient/jdbc/XGStatement.java [hunk 1]
```diff
@@ -2706,21 +2714,63 @@ private int setPrioritySQL(final String cmd) throws SQLException
 		final Matcher m = XGRegexUtils.genericSetSyntaxMatch("priority", cmd);
 		if (!m.matches())
 		{
-			throw SQLStates.SYNTAX_ERROR.cloneAndSpecify("Syntax error. Proper set priority syntax: set priority <priorityValue>");
+            throw SQLStates.SYNTAX_ERROR.cloneAndSpecify("Syntax error. Proper set priority syntax: set priority <priorityValue>");
```
Comment (tli-ocient) [inline src/main/java/com/ocient/jdbc/XGStatement.java:2717]: Nit: it seems you misaligned this

**Review by tli-ocient — CHANGES_REQUESTED** [review]
Some small changes needed. Also need to update the driver version in `pom.xml` and add a release note in `release_nodes.adoc`. 
To your point about testing.
CI will run against this version of JDBC when you try to release it to xgsrc. We can talk about that once this PR merges.

File: src/main/java/com/ocient/jdbc/XGConnection.java [hunk 2]
```diff
@@ -3064,7 +3067,16 @@ private void validateDefaultProperties(Properties properties) throws SQLExceptio
...
+		}
+
+		if (properties.containsKey("priorityAdjustTime") && properties.get("priorityAdjustTime") != null)
+		{
+			int proposedPriorityAdjustTime = Integer.parseInt((String) properties.get("priorityAdjustTime"));
+			if(proposedPriorityAdjustTime < 0){
+				throw SQLStates.INVALID_ARGUMENT.cloneAndSpecify(String.format("priority_adjustment_time must be >= 0, specified: %d", proposedPriorityAdjustTime));
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (susmita-saha) [reply src/main/java/com/ocient/jdbc/XGConnection.java:3079]: No, not right now.

File: src/main/java/com/ocient/jdbc/XGStatement.java [hunk 3]
```diff
@@ -2706,21 +2714,63 @@ private int setPrioritySQL(final String cmd) throws SQLException
...
-			priority = Integer.parseInt(ending);
+			priority = Double.parseDouble(ending);
 		}
 		catch (final NumberFormatException e)
 		{
 		}
 		return conn.setPriority(priority, reset);
 	}
 
+	private int setPriorityAdjustFactorSQL(final String cmd) throws SQLException
```
[the remark this answers is not in the document]
Comment (susmita-saha) [reply src/main/java/com/ocient/jdbc/XGStatement.java:2732]: yes, it is already added, but it is in XGConnection. See above.

File: src/main/java/com/ocient/jdbc/XGStatement.java [hunk 4]
```diff
@@ -2713,14 +2721,56 @@ private int setPrioritySQL(final String cmd) throws SQLException
...
+	{
+		LOGGER.log(Level.INFO, "Entered driver's setPriorityAdjustFactor()");
+		final Matcher m = XGRegexUtils.genericSetSyntaxMatch("adjustfactor", cmd);
+		if (!m.matches())
+		{
+            throw SQLStates.SYNTAX_ERROR.cloneAndSpecify("Syntax error. Proper syntax: set adjustfactor <priorityAdjFactor>");
```
Comment (tli-ocient) [inline src/main/java/com/ocient/jdbc/XGStatement.java:2738]: Nit: one more alignment issue

File: userdoc/release_notes.adoc [hunk 5]
```diff
@@ -1021,4 +1021,5 @@ New Features::
   * https://jira.ocient.com:8443/browse/DB-10137[DB-10137] - Implement JDBC cancel query. Now it is possible to kill query in dbeaver.
   * https://jira.ocient.com:8443/browse/DB-10120[DB-10120] - JDBC driver support list of sql nodes to connect to
   * https://jira.ocient.com:8443/browse/DB-10119[DB-10119] - Support for multiple IPs under the same DNS address
+  * https://ocient.atlassian.net/browse/DB-19576[DB-19576] - Added CLI support for SET ADJUSTFACTOR, ADJUSTTIME
```
Comment (tli-ocient) [inline userdoc/release_notes.adoc:1024]: No, at the top of this file add this:

`
//tag::compact[]

== 2.54 (2022-3-9)

New Features::

 * Added CLI support for SET ADJUSTFACTOR, ADJUSTTIME
`

Comment (tli-ocient) [reply userdoc/release_notes.adoc:1024]: Also change the version in `pom.xml` to `2.54`

File: userdoc/release_notes.adoc [hunk 6]
```diff
@@ -4,6 +4,14 @@
 All our jdbc drivers are located {drivers_repo}[here]. 
 Below is the release notes for every driver version that has customer implication.
 
+// tag::compact[]
+
+== 2.54 (2022-3-9)
+
+New Features::
+
+  * https://ocient.atlassian.net/browse/DB-19576[DB-19576] - Added CLI support for SET ADJUSTFACTOR, ADJUSTTIME
```
Comment (tli-ocient) [inline userdoc/release_notes.adoc:13]: These release notes might be seen by customers. Please remove the ticket

**Review by tli-ocient — APPROVED** [review]

**Pull request merged**